src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2738,7 +2739,7 @@ private bool IsMisplacedModifier(SyntaxListBuilder modifiers, SyntaxList<Attribu
                 }
 
                 var incompleteMember = _syntaxFactory.IncompleteMember(attributes, modifiers.ToList(), type.IsMissing ? null : type);
-                if (incompleteMember.ContainsDiagnostics)
+                if (incompleteMember.GetDiagnostics().Any(static d => d.Severity == DiagnosticSeverity.Error))
```

Comment:
Nice!

Reply:
@CyrusNajmabadi The following currently compiles due to [another use](https://github.com/dotnet/roslyn/blob/d4b6140d5ea3642f9696ff84ec402fb651a12fe1/src/Compilers/CSharp/Portable/Parser/LanguageParser.cs#L558) of `ContainsDiagnostics`:

```csharp
public class C
{
    public void M()
    {
    }
}
#warning test
]
```

I guess we should revise every use of ContainsDiagnostics in the parser.

Reply:
Yes.  Most are incorrect.  That's out of scope of what i'm trying to do here.

Reply:
@CyrusNajmabadi Agreed. I'll open a separate issue since this PR will close https://github.com/dotnet/roslyn/issues/13719